scripts/deployCollector.ts

```diff
@@ -0,0 +1,159 @@
+import * as fs from 'fs';
+
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { Collector } from 'typechain-types';
+
+export const DEFAULT_CONFIG_FILE_NAME = 'deploy-collector.input.json';
+export const DEFAULT_OUTPUT_FILE_NAME = 'revenue-sharing-addresses.json';
+
+export interface Partner {
+  beneficiary: string;
+  share: number;
+}
+
+interface CollectorConfig {
+  collectorOwner: string;
+  partners: Partner[];
+  tokenAddress: string;
+  remainderAddress: string;
+}
+
+export type DeployCollectorArg = {
+  collectorConfigFileName?: string;
+  outputFileName?: string;
+};
+
+type ChainId = string;
+type OutputCollectorConfig = CollectorConfig | { collectorContract: string };
+export type OutputConfig = Record<ChainId, OutputCollectorConfig>;
+
+export const deployCollector = async (
+  taskArgs: DeployCollectorArg,
+  hre: HardhatRuntimeEnvironment
+) => {
+  const { ethers, network } = hre;
+
+  if (!network) {
+    throw new Error('Unknown Network');
+  }
+
+  let collectorConfigFileName = taskArgs.collectorConfigFileName;
+
+  if (!collectorConfigFileName) {
```

Comment:
Its not simple to call it, configFileName?
```suggestion
  if (!configFileName) {
```